app/controllers/users_controller.rb

```diff
@@ -86,6 +86,39 @@ def activityhistory
     @earn_transactions = @earn_transactions.where('created_at <= ?', Date.parse(params[:end_date])) if params[:end_date].present?
   end
 
+  def history
+    earn = EarnTransaction.all
+    spend = SpendTransaction.all
+    @history = []
+
+    earn.each do |e|
+      activity = Activity.find_by(id: e.activity_id)
+      user = User.find_by(id: e.user_id)
+
+      if activity && user
+        #            Type,   user id,  user name,  activity id,  activity name,     points,            created at,      updated at
+        @history << ["Earned",user.id, user.name, activity.id, activity.name, "+#{e.points}", e.created_at, e.updated_at]
+      else
+        break
+      end
+
+    end
+
+    spend.each do |s|
+      reward = Reward.find_by(id: s.reward_id)
+      user = User.find_by(id: s.user_id)
+
+      if reward && user
+        #             Tyoe  ,user id,   user name,  reward id, reward name,         points,     created at,   updated at
+        @history << ["Spent", user.id, user.name, reward.id, reward.name, -1 * reward.point_value, s.created_at, s.updated_at]
```

Comment:
As we discussed:
In this line, multiplying the point value by −1 creates an invalid value. The minus should just be tacked onto the left when creating the string.